app/src/main/java/eu/kanade/tachiyomi/data/backup/restore/restorers/ExtensionRepoRestorer.kt

```diff
@@ -0,0 +1,37 @@
+package eu.kanade.tachiyomi.data.backup.restore.restorers
+
+import eu.kanade.tachiyomi.data.backup.models.BackupExtensionRepos
+import mihon.domain.extensionrepo.interactor.GetExtensionRepo
+import tachiyomi.data.DatabaseHandler
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
+
+class ExtensionRepoRestorer(
+    private val handler: DatabaseHandler = Injekt.get(),
+    private val getExtensionRepos: GetExtensionRepo = Injekt.get()
+) {
+
+    suspend operator fun invoke(backupExtensionRepos: List<BackupExtensionRepos>) {
+        if (backupExtensionRepos.isNotEmpty()) {
+            val dbExtensionRepos = getExtensionRepos.getAll()
+            val dbExtensionReposByName = dbExtensionRepos.associateBy { it.name }
+
+            backupExtensionRepos
+                .sortedBy { it.signingKeyFingerprint }
+                .forEach { backupRepo ->
+                    val dbExtensionRepo = dbExtensionReposByName[backupRepo.name]
```

Comment:
Should be by url. Also on that note we want to do something if signature clash while url is different.

Reply:
Is adding a log in the errors okay? 
I don't know how common it would be to have conflicting keys but I feel like having a pop up warning you about the conflict like when adding from the repos screen might be too much while restoring.
![Screenshot_20240731_113237_HTML_Viewer](https://github.com/user-attachments/assets/895afe30-12e2-45b3-ba7f-20bae806519b)
![Screenshot_20240731_113200_HTML_Viewer](https://github.com/user-attachments/assets/351a098b-d142-4344-a615-05e73741ffe5)

Reply:
Switched the error to this. I would have added the name of the existing repo as well but I didn't feel like having detekt yell at me about max line lengths. 🤣
error("Extension Repo not added: ${backupRepo.name} has the same signature as an existing repo")